stdlib/optional-parens.kl

```diff
@@ -0,0 +1,74 @@
+#lang kernel
```

Comment:
renamed from `builtins.kl`, because

1. we're calling them "primitives" everywhere else
2. this file is more concerned with making parens optional around some primitives than with exporting all the primitives.

Reply:
In addition to the rename, this file now makes parentheses optional around many more primitives, most notably types with no type arguments such as `String`.